main.tf

```diff
@@ -92,24 +92,25 @@ resource "azurerm_template_deployment" "app_service_site" {
   deployment_mode     = "Incremental"
 
   parameters = {
-    name                   = "${var.product}-${var.env}${var.deployment_target}"
-    location               = "${var.location}"
-    env                    = "${var.env}${var.deployment_target}"
-    app_settings           = "${jsonencode(merge(local.production_slot_app_settings, var.app_settings_defaults, local.app_settings_evaluated, var.app_settings))}"
-    staging_app_settings   = "${jsonencode(merge(var.staging_slot_app_settings, var.app_settings_defaults, local.app_settings_evaluated, var.app_settings))}"
-    additional_host_name   = "${var.additional_host_name}"
-    stagingSlotName        = "${var.staging_slot_name}"
-    is_frontend            = "${var.is_frontend}"
-    https_only             = "${var.https_only}"
-    capacity               = "${var.capacity}"
-    instance_size          = "${var.instance_size}"
-    web_sockets_enabled    = "${var.web_sockets_enabled}"
-    asp_name               = "${local.asp_name}"
-    asp_rg                 = "${local.asp_rg}"
-    teamName               = "${lookup(var.common_tags, "Team Name")}"
-    java_version           = "${var.java_version}"
-    java_container_type    = "${var.java_container_type}"
-    java_container_version = "${var.java_container_version}"
+    name                    = "${var.product}-${var.env}${var.deployment_target}"
+    location                = "${var.location}"
+    env                     = "${var.env}${var.deployment_target}"
+    app_settings            = "${jsonencode(merge(local.production_slot_app_settings, var.app_settings_defaults, local.app_settings_evaluated, var.app_settings))}"
+    staging_app_settings    = "${jsonencode(merge(var.staging_slot_app_settings, var.app_settings_defaults, local.app_settings_evaluated, var.app_settings))}"
+    additional_host_name    = "${var.additional_host_name}"
+    stagingSlotName         = "${var.staging_slot_name}"
+    is_frontend             = "${var.is_frontend}"
+    https_only              = "${var.https_only}"
+    capacity                = "${var.capacity}"
+    instance_size           = "${var.instance_size}"
+    web_sockets_enabled     = "${var.web_sockets_enabled}"
+    asp_name                = "${local.asp_name}"
+    asp_rg                  = "${local.asp_rg}"
+    teamName                = "${lookup(var.common_tags, "Team Name")}"
+    java_version            = "${var.java_version}"
+    java_container_type     = "${var.java_container_type}"
+    java_container_version  = "${var.java_container_version}"
+    client_affinity_enabled = "${var.client_affinity_enabled}"
```

Comment:
This doesn't appear to be defined:
```
var.client_affinity_enabled
```

Also why are you doing this?

All applications are required to be stateless

There's a reason that value is hardcoded in the template, you aren't allowed to turn it on

Reply:
The issue is that for the Multi Factor authentication, currently our only option is to use an authentication chain in openam which stores the state of the transaction in memory, if we redirect back to the wrong box we wont be able to complete the authentication. This requires affinity from web-public to idam-api and from idam-api to openam. Technically we wouldn't need affinity down to idam-api if the standard load balancers allowed for cookie based routing. They only allow for IP based affinity so we need affinity on both ends of the path.

Reply:
This needs further discussing offline, cc @njrich28 

Slipping it into an HLD doesn't mean it can be done
This is a platform requirement and enforced by the platform that you can't be stateful

Reply:
> The issue is that for the Multi Factor authentication, currently our only option is to use an authentication chain in openam which stores the state of the transaction in memory, if we redirect back to the wrong box we wont be able to complete the authentication. This requires affinity from web-public to idam-api and from idam-api to openam. Technically we wouldn't need affinity down to idam-api if the standard load balancers allowed for cookie based routing. They only allow for IP based affinity so we need affinity on both ends of the path.

I assume you've tested this in preview with some manual changes?

What about the application gateway as well? I don't think it's configurable in the module there either, but haven't checked

Reply:
@timja We'd appreciate it if you make sure this conversation can happen ASAP due to the 1.5 release schedule.

Reply:
(I'm discussing with Kremena currently, we'll have a decision by end of tomorrow)

Reply:
There are 2 alternatives to enabling the affinity for our webapp:
1. Switch the LB in front of FR for an AppGw
2. Upgrade FR to a new and better version where things are not stored in memory (!) (I know, right)

We are doing 2 (coming some time in January) which effectively forces us to deploy as a temporary solution one of:
1. this PR
2. AppGw instead of an LB for FR

1 is preferred by the team due to less complexity and no change in infra design.